src/main/scala/scala/collection/next/NextIterableOnceOpsExtensions.scala

```diff
@@ -28,14 +48,54 @@ private[next] final class NextIterableOnceOpsExtensions[A, CC[_], C](
    *
    * @note This will force the evaluation of the Iterator.
    */
-  def groupMapReduce[K, B](key: A => K)(f: A => B)(reduce: (B, B) => B): immutable.Map[K, B] = {
-    val m = mutable.Map.empty[K, B]
-    col.foreach { elem =>
-      m.updateWith(key = key(elem)) {
-        case Some(b) => Some(reduce(b, f(elem)))
-        case None    => Some(f(elem))
+  def groupMapReduce[K, V](key: A => K)(f: A => V)(reduce: (V, V) => V): immutable.Map[K, V] =
+    groupMapGenGen(key)(f).reduceValues(reduce)
+}
+
+private[next] object NextIterableOnceOpsExtensions {
+  final class GroupMapGenGen[A, K, V] private[NextIterableOnceOpsExtensions](
+    col: IterableOnceOps[A, AnyConstr, _],
+    key: A => K,
+    f: A => V
+  ) {
+    def reduceValues(reduce: (V, V) => V): immutable.Map[K, V] =
+      reduceValuesAs(immutable.Map)(reduce)
```

Comment:
Is this different from `groupMapReduce`? Maybe it’s not necessary to have both?

Reply:
Not really, as you can see `groupMapReduce` is implemented in terms of this one...

Uhm, I think you are right, this method doesn't add any value. I should start proofreading my code before pushing 😅